lib/webauthn/attestation_statement/tpm/pub_area.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require "cose/key"
+require "tpm/constants"
+require "tpm/t_public"
+
+module WebAuthn
+  module AttestationStatement
+    class TPM < Base
+      class PubArea
+        BYTE_LENGTH = 8
+
+        COSE_TO_TPM_ALG = {
+          COSE::Algorithm.by_name("ES256").id => ::TPM::ALG_ECDSA,
+          COSE::Algorithm.by_name("RS256").id => ::TPM::ALG_RSASSA
+        }.freeze
+
+        COSE_TO_TPM_CURVE = {
+          COSE::Key::EC2::CRV_P256 => ::TPM::ECC_NIST_P256
+        }.freeze
+
+        TPM_TO_OPENSSL_HASH_ALG = {
+          ::TPM::ALG_SHA256 => "SHA256"
+        }.freeze
+
+        def initialize(data)
+          @data = data
+        end
+
+        def valid?(public_key, alg)
+          cose_key = COSE::Key.deserialize(public_key)
+
+          case cose_key
+          when COSE::Key::EC2
+            valid_ecc_key?(cose_key, alg)
+          when COSE::Key::RSA
+            valid_rsa_key?(cose_key, alg)
+          else
+            raise "Unsupported or unknown TPM key type"
+          end
+        end
+
+        def valid_name
+          [t_public.name_alg].pack("n") + digest
```

Comment:
This approach is what I understood should be done by reading W3C recommendation.
However, conformance tool fails with this approach. See discussion in https://github.com/fido-alliance/conformance-tools-issues/issues/396.

The following PR implements the other approach which apparently is inconsistent with the W3C recommendation, but passes the conformance tool test: https://github.com/fido-alliance/conformance-tools-issues/issues/396.

Reply:
Merged #164 into this branch. Going with the conformance tool approach for now.